lib/src/coap_client.dart

```diff
@@ -35,338 +55,353 @@ class CoapClient {
   /// If it is specified it will be prepared and used.
   /// Note that the host name part of the URI can be a name or an IP address,
   /// in which case it is not resolved.
-  CoapClient(this.uri, this._config, [this.request]);
-
-  final CoapILogger? _log = CoapLogManager().logger;
-  static final Iterable<CoapWebLink> _emptyLinks = <CoapWebLink>[
-    CoapWebLink('')
-  ];
-
-  /// The URI
-  Uri uri;
-  final DefaultCoapConfig _config;
-
-  String get _namespace =>
-      _config.poolUdpConnectionsByClient ? hashCode.toString() : '';
-
-  /// The endpoint. Once set, on the first request this endpoint is used
-  /// throughout this client for all subsequent requests.
-  CoapIEndPoint? endpoint;
-
-  int _type = CoapMessageType.con;
-  int _blockwise = 0;
-
-  /// The current request being processed by the client.
-  CoapRequest? request;
-
-  /// Timeout
-  int timeout = 32767; //ms
+  CoapClient(this.uri, this._config,
+      {this.addressType = InternetAddressType.IPv4, Duration? timeout}) {
+    this.timeout = timeout ?? Duration(milliseconds: _config.defaultTimeout);
+    _namespace = hashCode.toString();
+  }
 
   /// Address type used for DNS lookups.
-  InternetAddressType addressType = InternetAddressType.IPv4;
+  final InternetAddressType addressType;
 
-  /// Tell the client to use Confirmable requests.
-  CoapClient useCONs() {
-    _type = CoapMessageType.con;
-    return this;
-  }
+  /// The client endpoint URI
+  final Uri uri;
 
-  /// Let the client use early negotiation for the blocksize
-  /// (16, 32, 64, 128, 256, 512, or 1024). Other values will
-  /// be matched to the closest logarithm.
-  CoapClient useEarlyNegotiation(int size) {
-    _blockwise = size;
-    return this;
-  }
+  /// The default request timeout
+  late Duration timeout; // ms
 
-  /// Let the client use late negotiation for the block size (default).
-  CoapClient useLateNegotiation() {
-    _blockwise = 0;
-    return this;
-  }
+  final DefaultCoapConfig _config;
+  CoapIEndPoint? _endpoint;
+  late final String _namespace;
```

Comment:
Yeah, the reason I did it this way was to avoid repeated type conversions, but I can change, I'm not strongly in favor of either approach and it's a tiny cost.

Reply:
Fixed: https://github.com/shamblett/coap/pull/76/commits/23f12832be6d414080a70c4193c1a4afe34c44e9